### PR TITLE
Refactor2

### PR DESCRIPTION
**Pull request by Noboomta**
Add endpoint
- add **PATCH** add people/add/:date

Refactor and add more comment in people route
- refactor **GET** /people/all
- refactor **GET** people/by_date/:date
- refactor **DELETE** people/by_date/:date
- refactor **GET** people/count/total/:date
- refactor **GET** people/count/walkin/:date
- refactor **PATCH** people/cancel/:date/:reservationID
- refactor **PATCH** people/vaccinated/:date/:reservationID
- refactor **patch** people/add/:date

### CODE REVIEW

**Comment by Noboomta** [comment]
Please read most of commited.

**Review by lisbono2001 — APPROVED** [review]
Excellent✌️

**Review by toey10112 — APPROVED** [review]

**Review by patiphan2000 — APPROVED** [review]

File: src/routes/people.route.js [hunk 1]
```diff
@@ -1,107 +1,254 @@
...
+	});
 
+	/**
+		GET people/by_date/:date
+		used: get people schema by date.
+		status code: 
+			- 200 OK
+			- 406 no date param included in request.
+	*/
+	router.get("/by_date", "/by_date/:date", async (req, res) => {
```
Comment (toey10112) [inline src/routes/people.route.js:25]: should be like	router.get(["/by_date", "/by_date/:date"], async (req, res) => {

**Review by toey10112 — CHANGES_REQUESTED** [review]

File: src/routes/people.route.js (hunk 1 above)
Comment (Noboomta) [reply src/routes/people.route.js]: Thankyou!!

**Review by toey10112 — APPROVED** [review]
okay

File: src/routes/people.route.js [hunk 2]
```diff
@@ -15,16 +15,25 @@ function peopleRoutes(database) {
...
 		console.log("get people by date");
 		console.log(req.params);
+		const { date } = req.params;
+		if (!date) {
+			return res.status(406).json({ msg: "no date param included" });
+		}
+		
 		const people = await database.getPeopleInfoByDate({
-			date: req.params.date,
+			date: data
```
Comment (patiphan2000) [inline src/routes/people.route.js:34]: should it be `date: date` ?

Comment (Noboomta) [reply src/routes/people.route.js]: Ok thankyou. sorry...

**Comment by Noboomta** [comment]
Thank to everyone, It seems like everything is correct and works now.

**Comment by lisbono2001** [comment]
Guys there are just some more details added

**Review by lisbono2001 — APPROVED** [review]

**Comment by Noboomta** [comment]
> Guys there are just some more details added
Big "Finally" XD

**Pull request merged**